gnmidiff/setrequest_test.go

```diff
@@ -20,10 +20,543 @@ import (
+			}},
+		},
+		inB: &gpb.SetRequest{
+			Replace: []*gpb.Update{{
+				Path: ygot.MustStringToPath("/interfaces/interface[name=eth0]"),
+				Val:  must7951(&exampleoc.Interface{Name: ygot.String("eth0"), Transceiver: ygot.String("FDM")}),
+			}, {
+				Path: ygot.MustStringToPath("/interfaces/interface[name=eth2]"),
+				Val:  must7951(&exampleoc.Interface{Name: ygot.String("eth2"), Transceiver: ygot.String("FDM")}),
+			}},
+			Update: []*gpb.Update{{
+				Path: ygot.MustStringToPath("/interfaces/interface[name=eth0]"),
+				Val:  must7951(&exampleoc.Interface{Subinterface: map[uint32]*exampleoc.Interface_Subinterface{0: {Index: ygot.Uint32(0), OperStatus: exampleoc.Interface_OperStatus_TESTING, Enabled: ygot.Bool(true)}}}),
+			}, {
+				Path: ygot.MustStringToPath("/interfaces/interface[name=eth0]"),
+				Val:  must7951(&exampleoc.Interface{Description: ygot.String("I am an eth port")}),
+			}, {
+				Path: ygot.MustStringToPath("/interfaces/interface[name=eth0]/subinterfaces/subinterface[index=0]/state/logical"),
+				Val:  &gpb.TypedValue{Value: &gpb.TypedValue_BoolVal{BoolVal: true}},
+			}},
+		},
+		wantSetRequestDiff: SetRequestIntentDiff{
+			AOnlyDeletes: map[string]struct{}{
+				"/interfaces/interface[name=eth1]": {},
+			},
+			BOnlyDeletes: map[string]struct{}{
+				"/interfaces/interface[name=eth2]": {},
+			},
+			CommonDeletes: map[string]struct{}{
+				"/interfaces/interface[name=eth0]": {},
+			},
+			AOnlyUpdates: map[string]interface{}{
+				"/interfaces/interface[name=eth1]/name":        "eth1",
+				"/interfaces/interface[name=eth1]/config/name": "eth1",
+			},
+			BOnlyUpdates: map[string]interface{}{
+				"/interfaces/interface[name=eth2]/name":              "eth2",
+				"/interfaces/interface[name=eth2]/config/name":       "eth2",
+				"/interfaces/interface[name=eth0]/state/transceiver": "FDM",
+			},
+			CommonUpdates: map[string]interface{}{
+				"/interfaces/interface[name=eth0]/name":                                                  "eth0",
+				"/interfaces/interface[name=eth0]/config/name":                                           "eth0",
+				"/interfaces/interface[name=eth0]/config/description":                                    "I am an eth port",
+				"/interfaces/interface[name=eth0]/subinterfaces/subinterface[index=0]/config/index":      float64(0),
+				"/interfaces/interface[name=eth0]/subinterfaces/subinterface[index=0]/index":             float64(0),
+				"/interfaces/interface[name=eth0]/subinterfaces/subinterface[index=0]/state/oper-status": "TESTING",
+				"/interfaces/interface[name=eth0]/subinterfaces/subinterface[index=0]/config/enabled":    true,
+				"/interfaces/interface[name=eth2]/state/transceiver":                                     "FDM",
+			},
+			MismatchedUpdates: map[string]MismatchedUpdate{
+				"/interfaces/interface[name=eth0]/subinterfaces/subinterface[index=0]/state/logical": {
+					A: false,
+					B: true,
+				},
+			},
+		},
+	}}
```

Comment:
Did you have any "duplicate but out of order" updates where the sorted intent looks the same but the results are actually different?  I think the stuffing these into a map in order will address that but it would be nice to be sure.

Reply:
I'm not 100% sure what you mean by "duplicate but out of order", but right now conflicting leaf updates produce an error. This limitation can be removed later. Once it's removed, the update that occurs later will replace the previous one via the map key, as you've mentioned.

Reply:
I was meaning that set request A has two updates for the same node (say node X) with the first update setting it to Y, and the second to Z.  Request B also has two updates for node X, but first setting it to Z and then setting it to Y.  Request A will result in node X being Z and request B will result in node X being Y.  So both requests have the same two update requests for X but they are in different orders in each request.  Perhaps this is not possible, I don't know.

Reply:
This is possible, and the general solution which this implementation is intended to be adaptable towards will capture this by over-writing the new value of node X with the last update seen in the SetRequest. It's just not in this MVP.